crates/ibc/src/core/context.rs

```diff
@@ -292,8 +292,19 @@ pub trait ValidationContext: Router {
     /// Returns the ConnectionEnd for the given identifier `conn_id`.
     fn connection_end(&self, conn_id: &ConnectionId) -> Result<ConnectionEnd, ContextError>;
 
-    /// Validates the `ClientState` of the client on the counterparty chain.
-    fn validate_self_client(&self, counterparty_client_state: Any) -> Result<(), ConnectionError>;
+    /// Validates the client state of the host chain that is stored on the
+    /// counterparty chain.
+    ///
+    /// For more information on the specific requirements for validating the
+    /// client state of a host chain, please refer to the [ICS24 host
+    /// requirements](https://github.com/cosmos/ibc/tree/main/spec/core/ics-024-host-requirements#client-state-validation)
+    ///
+    /// Additionally, implementations specific to individual chains can be found
+    /// in the [hosts](crate::hosts) module.
+    fn validate_self_client(
+        &self,
+        host_client_state_on_counterparty: Any,
```

Comment:
```suggestion
        client_state_of_host_on_counterparty: Any,
```
This is closer to the convention we use in the rest of the codebase

Reply:
[907f6b2](https://github.com/cosmos/ibc-rs/pull/436/commits/907f6b24a815a8f2fa33f4aa702b67d791f25105)